docs/architecture/adr-004-application-links-expiration.md

```diff
@@ -0,0 +1,249 @@
+# ADR 004: Expiration of application links
+
+## Changelog
+
+- September 20th, 2021: Initial draft;
+- September 21th, 2021: Moved from DRAFT to PROPOSED;
+- December  22th, 2021: First review;
+- January   04th, 2022: Second review.
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+Currently when a user links their centralized application with the Desmos profile, the created link contains a timestamp of when it has been created.   
+Since centralized applications' username can be switched and sold, we SHOULD implement an "expiration date" system on links. 
+This means that after a certain amount of time passes, the link will be automatically marked as invalid, and the user has to connect it again in order to keep it valid.
+
+## Context
+
+Desmos `profiles` module give the possibility to link the desmos profile with both centralized application and 
+other blockchains accounts. By doing this, a user can be verified as the owner of those accounts and prove to the system
+that they're not impersonating anyone else. This verification however remains valid only if the user
+never trade or sell his centralized-app username to someone else. If they do the link to such username MUST be invalidated. 
+Unfortunately for us, it's not possible to understand when this happens since it's off-chain. 
+To prevent this situation, an "expiration time" SHOULD be added to the centralized `ApplicationLink` object.
+
+## Decision
```

Comment:
I'm honestly not a big fan of all this code inside the `Decisions` section. The more code we include, the more probabilities there are that we later need to change this ADR if we find better alternatives during its implementation. 

For this reason, I think we should instead describe what we should do on a high level. Something like this: 

```
In order to implement application link expiration, we will act as follow: 
1. Extend the `ApplicationLink` object by adding an expiration time field, which identifies the exact time on which the link will expire and thus SHOULD be deleted from the store 
2. Create a new `Keeper` method that is going to be called at the beginning of each block and delete all the expired application links. 

Also, in order to provide an easy way for the community to choose how long application links should be considered valid, we will also add the new params structure: 

...
```